README.md

```diff
@@ -92,14 +92,219 @@ class Genome(ABC):
+
+First, let's start with the List implementation. 
+
+### Init instance
+
+```python
+def __init__(self, n: int):
+    self.genome = n*['-'] # O(n)
+    self.active_identifiers = {} # O(1)
+    self.te_counter = 0# O(1)
+```
+
+Since appending to a list has a complexity of $O(1)$, creating the genome costs $O(n)$, whereas $n$ is the length of the initial genome. 
+
+### Insert a transposable element to the list
+
+```python
+def insert_te(self, pos: int, length: int) -> int:
+    pos = pos % len(self) # O(1)
+    active_identifiers = list(self.active_identifiers.items()) # O(k)
+    for identifier,active_te  in active_identifiers: # O(k)
+        is_in_range(pos, active_te): # O(1)
+            start, end = active_te # O(1)
+            self.genome[start:end] = (end - start)*["x"] # O(n + m)
+            self.active_identifiers.pop(identifier) # O(1)
+        if active_te.start > pos: O(1)
+            self.active_identifiers[identifier] = Interval(
+                active_te.start + length, active_te.end + length
+            ) # O(1)
+    self.genome[pos:pos] = length*['A'] # O(n + m)
+    self.te_counter += 1 # O(1)
+    self.active_identifiers[self.te_counter] = Interval(pos, pos + length) # O(1)
+    return self.te_counter
+```
+
+As you can see in the previous block, the final complexity is $O(k(n+m))$. However, in the worst case, all transposable elements are of length one and $k = n$, then we have $O(n^2\cdot m)$
+
+### Copy a transposable element
+
+```python
+def copy_te(self, te: int, offset: int) -> int | None:
+    original: Interval = self.active_identifiers.get(te) # O(1)
+    if original: O(1)
+        pos = (original.start + offset) # O(1)
+        length = original.end - original.start # O(1)
+        return self.insert_te(pos, length) # inserting transposable element
+```
+
+As we are using a dict (hash table) for storing the exact position of each active transposon, this operation is, after calculating the new position which can be done in $O(1)$, just inserting a transposable element. Then, in the worst case its complexity is the same as inserting an element and in the best case is $O(1)$. 
+
+### Disable a transposable element
+
+```python
+def disable_te(self, te: int) -> None:
+    original: Interval = self.active_identifiers.pop(te) # O(1)
+    if original: O(1)
+        self.genome[original.start:original.end] = (original.end - original.start)*["x"] # O(n+m)
+```
+
+Again, we can find the right indexes in $O(1)$. For disabling the te, we have to set a slice which, as [TimeComplexity Wiki](https://wiki.python.org/moin/TimeComplexity) says, its amortized worst case is $O(n, m)$.
+
+## Get active in te
+
+```python
+def active_tes(self) -> list[int]:
+    return list(self.active_identifiers.keys())
+```
+
+As we save all active te and update those entries in a dict in the other operations, getting that list is just $O(1)$. Or, if we copy it, just $O(k)$. 
```

Comment:
You will copy it here, when you translate the keys to a `list`. The complexity $O(k)$ is correct.